### PR TITLE
fix: handle failure to get access token

### PR DESCRIPTION
**Pull request by abutterworth**
ARCH-1066: Sometimes we fail to get an access token. Handle this case by showing our error page and add logging for future investigation.

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -73,13 +74,17 @@ apiClient.ensurePublicOrAuthenticationAndCookies(
...
     const { store, history } = configure();
 
-    ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
-
     if (accessToken) {
+      ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
       identifyAuthenticatedUser(accessToken.userId);
+      sendPageEvent();
     } else {
+      // This should never happen, but it does sometimes.
```
Comment (albemarle) [inline src/index.jsx]: :D

File: src/index.jsx [hunk 2]
```diff
@@ -73,13 +74,17 @@ apiClient.ensurePublicOrAuthenticationAndCookies(
...
+      // Add logging to learn more.
+      ReactDOM.render(<ErrorPage />, document.getElementById('root'));
+      NewRelicLoggingService.logError('Empty accessToken returned from ' +
+        'ensurePublicOrAuthenticationAndCookies callback.');
```
Comment (albemarle) [inline src/index.jsx]: Is there anything else useful we can add about the state?

File: src/index.jsx [hunk 3]
```diff
@@ -73,13 +74,17 @@ apiClient.ensurePublicOrAuthenticationAndCookies(
...
-    ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
-
     if (accessToken) {
+      ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
       identifyAuthenticatedUser(accessToken.userId);
+      sendPageEvent();
     } else {
+      // This should never happen, but it does sometimes.
+      // Add logging to learn more.
+      ReactDOM.render(<ErrorPage />, document.getElementById('root'));
```
Comment (robrap) [inline src/index.jsx]: Is there a way to just use ErrorBoundary?

Comment (abutterworth) [reply src/index.jsx]: We're handling this error outside the scope of React, so not directly no.

File: src/index.jsx (hunk 2 above)
Comment (abutterworth) [reply src/index.jsx]: @robrap Is there anything useful here? I can't think of anything.

Comment (robrap) [reply src/index.jsx]: @albemarle: FYI: We've got other related frontend-auth code changes to help log and track this down.
@abutterworth: We should think about the right interface for ensurePublicOrAuthenticationAndCookies though.  It seems wrong to be calling this callback in this error condition.

**Pull request merged**